src/pattern_matching/myers/long.rs

```diff
@@ -219,7 +219,7 @@ where
 
         let w = word_size::<T>();
         let last_dist = self.states[last_block].dist;
-        if (last_dist as i8 - carry as i8) as usize <= max_dist
+        if (last_dist as isize - carry as isize) as usize <= max_dist
```

Comment:
@markschl I needed to change this because the example in the test below creates an overflow otherwise.

1. Can the difference above become negative (e.g. with last_dist = 0 and carry = 1?).
2. What values are possible for carry at all? I have seen -1, which suprised me a bit (but I did not dig into the algorithm enough to judge).

Reply:
I think, carry is only ever supposed to have the values -1, 0 or 1, if not there may be a bug somewhere. I will have a look.

Reply:
Could it happen that last_dist is zero and carry is 1? Leading to -1? Then, the usize cast would lead to a very large number.

Reply:
I had a look at the problem, and I think changing to `i8` to `isize` is completely fine and will resolve the problem.

`last_dist` is the distance at the last ("lowermost" position of the last block in the computed zone of the DP matrix. In this case, `last_dist = 127`, `carry = -1`, and calculating `127 as i8 - -1 as i8` will lead to an overflow. This effectively limits pattern length to 127, I think. 

I'm sorry that I overlooked this obvious problem. Since the same tests are implemented for both the 'simple' and the 'long' version of the algorithm, and the 'simple' version cannot represent such long patterns, this problem went undetected. In the fuzzing code, I [limited possible pattern length to 64](https://github.com/rust-bio/rust-bio/blob/789c55865cc5acc6828f2aa468e963ba7546e199/fuzz/fuzz_targets/myers_matching.rs#L18) in order to be able to compare the results of both algorithm versions. This restriction could be removed by adding another function, which tests only `long::Myers` for longer patterns (as well as possible given that the 'ground truth' is not exactly known in a random pattern).

Reply:
Ok, thank you! I am happy to take additional PRs on this. For now, seems fine to merge.